src/ImageFitting/ImageFitter.cc

```diff
@@ -48,7 +48,36 @@ bool ImageFitter::FitImage(size_t width, size_t height, float* image, double bea
     _progress_callback = progress_callback;
 
     CalculateNanNumAndStd();
-    SetInitialValues(initial_values, background_offset, fixed_params);
+    success = SetInitialValues(initial_values, background_offset, fixed_params);
+
+    std::string initial_value_log = "";
+    if (!success) {
+        spdlog::info("Generating initial values for fitting.");
+        InitialValueCalculator* calculator = new InitialValueCalculator(&_fit_data, initial_values, _image_std);
```

Comment:
Memory leak--`new` but no `delete`, use unique_ptr

Reply:
Good point; I missed this. But why is this a pointer at all? It could just be a normal object variable.

Reply:
That's true, it does not get passed anywhere.